modular_chomp/code/modules/mob/living/simple_mob/subtypes/humanoid/eclipse/heads.dm

```diff
@@ -531,31 +541,142 @@
+			potential_targets -= target
+			spawn_lines(target)
+
+
+/mob/living/simple_mob/humanoid/eclipse/proc/spawn_lines(atom/target)
+	var/alignment = rand(1,2)	// 1 for vertical, 2 for horizontal
+	var/list/line_range = list()
+	var/turf/T = get_turf(target)
+	line_range += T
+	for(var/i = 1, i <= 7, i++)
+		switch(alignment)
+			if(1)
+				if(T.x-i > 0)
+					line_range += locate(T.x-i, T.y-i, T.z)
+				if(T.x+i <= world.maxx)
+					line_range += locate(T.x+i, T.y+i, T.z)
+				if(T.y-i > 0)
+					line_range += locate(T.x+i, T.y-i, T.z)
+				if(T.y+i <= world.maxy)
+					line_range += locate(T.x-i, T.y+i, T.z)
+			if(2)
+				if(T.x-i > 0)
+					line_range += locate(T.x-i, T.y, T.z)
+				if(T.x+i <= world.maxx)
+					line_range += locate(T.x+i, T.y, T.z)
+				if(T.y-i > 0)
+					line_range += locate(T.x, T.y-i, T.z)
+				if(T.y+i <= world.maxy)
+					line_range += locate(T.x, T.y+i, T.z)
+	for(var/turf/dropspot in line_range)
+		new /obj/effect/artillery_attack(dropspot)
+
+
+/mob/living/simple_mob/humanoid/eclipse/proc/bomb_chaos(atom/A)
+	var/list/potential_targets = ai_holder.list_targets()
+	for(var/atom/entry in potential_targets)
+		if(istype(entry, /mob/living/simple_mob/humanoid/eclipse))
+			potential_targets -= entry
+	if(potential_targets.len)
+		var/iteration = clamp(potential_targets.len, 1, 3)
+		for(var/i = 0, i < iteration, i++)
+			if(!(potential_targets.len))
+				break
+			var/mob/target = pick(potential_targets)
+			potential_targets -= target
+			chaos_lines(target)
+
+
+/mob/living/simple_mob/humanoid/eclipse/proc/chaos_lines(atom/target)
+	var/alignment = rand(1,2)
+	var/list/line_range = list()
+	var/turf/T = get_turf(target)
+	line_range += T
+	for(var/i = 1, i <= 7, i++)
+		switch(alignment)
+			if(1)
+				if(T.x-i > 0)
+					var/zed = rand(1,3)
+					line_range += locate(T.x+zed, T.y-i, T.z)
+				if(T.x+i <= world.maxx)
+					var/zed = rand(1,3)
+					line_range += locate(T.x+zed, T.y+i, T.z)
+				if(T.y-i > 0)
+					var/zed = rand(1,3)
+					line_range += locate(T.x+i, T.y+zed, T.z)
+				if(T.y+i <= world.maxy)
+					var/zed = rand(1,3)
+					line_range += locate(T.x-i, T.y+zed, T.z)
+			if(2)
+				if(T.x-i > 0)
+					var/zed = rand(1,3)
+					line_range += locate(T.x-i, T.y-zed, T.z)
+				if(T.x+i <= world.maxx)
+					var/zed = rand(1,3)
+					line_range += locate(T.x+i, T.y-zed, T.z)
+				if(T.y-i > 0)
+					var/zed = rand(1,3)
+					line_range += locate(T.x-zed, T.y-i, T.z)
+				if(T.y+i <= world.maxy)
+					var/zed = rand(1,3)
+					line_range += locate(T.x-zed, T.y+i, T.z)
+	for(var/turf/dropspot in line_range)
+		new /obj/effect/artillery_attack(dropspot)
+
+
+/obj/effect/artillery_attack
+	anchored = TRUE
+	density = FALSE
+	unacidable = TRUE
+	mouse_opacity = 0
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "drop_marker"
+
+/obj/effect/artillery_attack/Initialize(mapload)
+	..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/effect/artillery_attack/LateInitialize()
+	var/delay = rand(25, 30)
+	spawn(delay-7)
```

Comment:
could also be callbacked

I also wouldn't subtract and rather add on the deletion proc, it gives more freedom to play with numbers as you can't go negative